nominatim/tools/collect_os_info.py

```diff
@@ -0,0 +1,161 @@
+"""
+Collection of host system information including software versions, memory,
+storage, and database configuration.
+"""
+import os
+from pathlib import Path
+import subprocess
+import sys
+from typing import Optional, Union
+
+# external requirement
+import psutil
+
+from nominatim.version import NOMINATIM_VERSION
+from nominatim.db.connection import connect
+
+
+class ReportSystemInformation:
+    """Generate a report about the host system including software versions, memory,
+    storage, and database configuration."""
+
+    def __init__(self, config):
+        self._memory: int = psutil.virtual_memory().total
+        self.friendly_memory: str = self._friendly_memory_string(self._memory)
+        # psutil.cpu_count(logical=False) returns the number of CPU cores.
+        # For number of logical cores (Hypthreaded), call psutil.cpu_count() or os.cpu_count()
+        self.num_cpus: int = psutil.cpu_count(logical=False)
+        self.os_info: str = self._os_name_info()
+
+        self.nominatim_ver: str = '{0[0]}.{0[1]}.{0[2]}-{0[3]}'.format(
+            NOMINATIM_VERSION)
```

Comment:
nominatim.version now has a convenience function `version_str()`.

Reply:
I globally searched this function and couldn't find it.

Reply:
I mean this one here: https://github.com/osm-search/Nominatim/blob/800240550bf30a306ca7b23568bab24e3a894920/nominatim/version.py#L40

Reply:
Yes, I have synced up the branch to the master branch because the branch was not updated for a long time and then found that function.